File: 9a.pl
#!/usr/bin/perl
use warnings;
use strict;
use feature qw{ say };

use ARGV::OrDATA;

my @heightmap;
while (<>) {
    chomp;
    push @heightmap, [split //];
}

my $risk_level = 0;
for my $y (0 .. $#heightmap) {
    for my $x (0 .. $#{ $heightmap[$y] }) {
        my $current =  $heightmap[$y][$x];
        next if $current == 9
            || $y > 0                    && $current >= $heightmap[$y - 1][$x]
            || $y < $#heightmap          && $current >= $heightmap[$y + 1][$x]
            || $x > 0                    && $current >= $heightmap[$y][$x - 1]
            || $x < $#{ $heightmap[$y] } && $current >= $heightmap[$y][$x + 1];

        $risk_level += $heightmap[$y][$x] + 1;
    }
}
say $risk_level;

__DATA__
2199943210
3987894921
9856789892
8767896789
9899965678

File: 9a.py
#! /usr/bin/python3
import sys

heightmap = []
with open(sys.argv[1], 'r') as fh:
    for line in fh:
        heightmap.append([c for c in line[:-1]])

risk_level = 0;
for y in range(len(heightmap)):
    for x in range(len(heightmap[y])):
        current = heightmap[y][x]
        if current == 9 \
           or y > 0 and current >= heightmap[y-1][x] \
           or y < len(heightmap)-1 and current >= heightmap[y+1][x] \
           or x > 0 and current >= heightmap[y][x-1] \
           or x < len(heightmap)-1 and current >= heightmap[y][x+1]:
            continue
        risk_level += int(heightmap[y][x]) + 1
print(risk_level)
